test/cut_and_merge/cut_and_merge_test.exs

```diff
@@ -0,0 +1,87 @@
+defmodule Membrane.VideoMerger.CutAndMergeTest do
+  use ExUnit.Case, async: true
+
+  import Membrane.ParentSpec
+  import Membrane.Testing.Assertions
+  alias Membrane.{Buffer, File, H264, Pad, Testing, Time}
+  alias Membrane.VideoMerger.CutAndMerge
+  alias Membrane.VideoMerger.CutAndMerge.Stream
+  require Pad
+
+  @fps 30
+  @framerate {@fps, 1}
+
+  defp run_pipeline_test(streams, test_length) do
+    assert {:ok, pid} = start_pipeline(streams)
+    assert(Testing.Pipeline.play(pid) == :ok)
+    frame_duration = Ratio.new(Time.second(), @fps)
+
+    assert_end_of_stream(pid, :sink, :input, 10_000)
+
+    for i <- 0..(test_length * @fps - 1) do
+      pts = Ratio.mult(frame_duration, i)
+      assert_sink_buffer(pid, :sink, %Buffer{metadata: %{pts: buffer_pts}})
+      assert pts == buffer_pts
+    end
+
+    Testing.Pipeline.stop_and_terminate(pid, blocking?: true)
+  end
+
+  defp start_pipeline(streams) do
+    elems = [merger_bin: CutAndMerge, sink: Testing.Sink]
+    links = [link(:merger_bin) |> to(:sink)]
+
+    {_i, elems, links} =
+      List.foldl(streams, {1, elems, links}, fn stream, {i, elems, links} ->
+        {src_i, parser_i, decoder_i} = {"file_src_#{i}", "parser_#{i}", "decoder_#{i}"}
+
+        new_elems = [
+          {src_i,
+           %File.Source{
+             chunk_size: 40_960,
+             location: "./test/fixtures/test_video_10s.h264"
+           }},
+          {parser_i, %H264.FFmpeg.Parser{framerate: @framerate}},
+          {decoder_i, H264.FFmpeg.Decoder}
+        ]
+
+        new_links = [
+          link(src_i)
+          |> to(parser_i)
+          |> to(decoder_i)
+          |> via_in(Pad.ref(:input, i), options: [stream: stream])
+          |> to(:merger_bin)
+        ]
+
+        {i + 1, elems ++ new_elems, links ++ new_links}
+      end)
```

Comment:
I get a feeling you're overusing `foldl` a bit:
1. Usually it's more common to use `Enum.reduce`
2. It's very generic and powerful - you can implement most of the functions in Enum using it - but I'd rather use it as a last resort
3. Here you don't really need to have elems and links at this stage 
4. Using an imperative counter like this in functional programming is a bit 'meh'

My suggestion (feel free to modify it and check if it's OK - it was written only here, on github without formatter nor compiler)
```suggestion
    elem_names =
      streams
      |> Enum.with_index(fn stream, i ->
         {stream, "file_src_#{i}", "parser_#{i}", "decoder_#{i}"}
      end)

    elems =
      elem_names
      |> Enum.flat_map(fn {_stream, src_i, parser_i, decoder_i} ->
        [
          {src_i,
           %File.Source{
             chunk_size: 40_960,
             location: "./test/fixtures/test_video_10s.h264"
           }},
          {parser_i, %H264.FFmpeg.Parser{framerate: @framerate}},
          {decoder_i, H264.FFmpeg.Decoder}
        ]
      end)

    links =
      elem_names
      |> Enum.map(fn {stream, src_i, parser_i, decoder_i} ->
          link(src_i)
          |> to(parser_i)
          |> to(decoder_i)
          |> via_in(Pad.ref(:input, i), options: [stream: stream])
          |> to(:merger_bin)
      end)
      
    elems = [merger_bin: CutAndMerge, sink: Testing.Sink | elems]
    links = [link(:merger_bin) |> to(:sink) | links]
```